parsers/test_json-aws_smithy_json/src/main.rs

```diff
@@ -32,7 +32,6 @@ fn main() {
                 match (token, current_open_token) {
                     // We can accept any starting container token, regardless of what we need to close
                     // IF we haven't completed a full container cycle (no forests)
-                    // PUSH token to vec
                     (&Token::StartArray { offset: _ } | &Token::StartObject { offset: _ }, _) => {
                         // multi-doc not allowed
                         if document_opened && document_closed {
```

Comment:
document_opened and document_closed is too much state, two booleans gives us four states. We have Initial, Opened, and Closed. We should use an enum for the three states.